step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java

```diff
@@ -342,7 +369,13 @@ protected List<Function> prepareFunctionsStaging(AutomationPackage newPackage, A
             completeFunctions.add(completeFunction);
         }
 
-        fillEntities(completeFunctions, oldPackage != null ? getPackageFunctions(oldPackage.getId()) : new ArrayList<>(), enricher);
+        // get old functions with same name and use their (i.e. update old functions)
+        List<Function> oldFunctions = new ArrayList<>();
+        for (Function newFunction : completeFunctions) {
+            oldFunctions.addAll(getFunctionsByAttributes(Map.of(AbstractOrganizableObject.NAME, newFunction.getAttribute(AbstractOrganizableObject.NAME))));
```

Comment:
@jeromecomte 
At that moment the layered accessors for keywords weren't used yet. And in this case we faced an issue with randomly chosen keyword (by name) during execution. So we tried to avoid creating keywords with same names across the whole system.

But the actual implementation in SED-2670 is
`oldFunctions = getPackageFunctions(oldPackage.getId())`